spec/types/object.spec.ts

```diff
@@ -93,6 +93,110 @@ describe('object', () => {
       });
     });
 
+    it('can chain-extend objects correctly', () => {
+      const BaseSchema = z.object({ id: z.string() }).openapi({
+        refId: 'Base',
+      });
+
+      const A = BaseSchema.extend({
+        bonus: z.number(),
+      }).openapi({
+        refId: 'A',
+      });
+
+      const B = A.extend({
+        points: z.number(),
+      }).openapi({
+        refId: 'B',
+      });
+
+      expectSchema([BaseSchema, A, B], {
+        Base: {
+          type: 'object',
+          required: ['id'],
+          properties: {
+            id: {
+              type: 'string',
+            },
+          },
+        },
+        A: {
+          allOf: [
+            { $ref: '#/components/schemas/Base' },
+            {
+              type: 'object',
+              required: ['bonus'],
+              properties: {
+                bonus: {
+                  type: 'number',
+                },
+              },
+            },
+          ],
+        },
+        B: {
+          allOf: [
+            { $ref: '#/components/schemas/A' },
+            {
+              type: 'object',
+              required: ['points'],
+              properties: {
+                points: {
+                  type: 'number',
+                },
+              },
+            },
+          ],
+        },
+      });
+    });
+
+    it('can chain-extend objects correctly without intermediate link', () => {
```

Comment:
This file is once again getting too big. I think it makes sense to move the tests for `.extend` in `modifiers/extends`. It is not exactly a modifier but it is pretty much there (we can change the name of the folder if need be).